contracts/dao/OptimisticTimelock.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+import {CoreRef} from "../refs/CoreRef.sol";
+
+// Timelock with veto admin roles
+contract OptimisticTimelock is TimelockController, CoreRef {
+    constructor(
+        address core_,
+        uint256 minDelay,
+        address[] memory proposers,
+        address[] memory executors
+    ) TimelockController(minDelay, proposers, executors) CoreRef(core_) {
+        // Only guardians and governors are timelock admins
+        revokeRole(TIMELOCK_ADMIN_ROLE, msg.sender);
+    }
+
+    /**
+        @notice allow guardian or governor to assume timelock admin roles
+        This more elegantly achieves optimistic timelock as follows:
+        - veto: grant self PROPOSER_ROLE and cancel
```

Comment:
Nit: This comment would need updating to be accurate as you updated the `@openzeppelin/contracts` version